File: src/api/crudSocio.js
const fs = require('fs');
const readline = require('readline');

const ARQUIVO = 'socios.json';

function carregarSocio() {
  if (!fs.existsSync(ARQUIVO))
    return[];
  return JSON.parse(fs.
  readFileSync(ARQUIVO));
}

function salvarSocio(socios) {
  fs.writeFileSync(ARQUIVO, JSON.
  stringify(socios, null, 2));
}

function listarSocio(socios, cb) {
  console.log('lista de socios:');
  socios.forEach(c => {
    console.log(`ID: ${c.id} | nome: ${c.nome} | email: ${c.email}`);
  });
  cb();
}

function addSocio(socios, rl, cb) {
  rl.question('Nome: ', nome => {
    rl.question('Email: ', email => {
      rl.question('Senha: ', senha => {
        const novoId = socios.length ? Math.max(...socios.map(c => c.id)) + 1 : 1;
        socios.push({ id: novoId, nome, email, senha});
        salvarSocio(socios);
        console.log('Socio adicionado.');
        cb();
      });
    });
  });
}

function editarSocio(socios, rl, cb) {
  rl.question('ID do socio para editar: ', idStr => {
    const id = parseInt(idStr);
    const socio = socios.find(c => c.id === id);
    if (!socio) {
      console.log('Socio inexistente.');
      return cb()
    }
      rl.question(`Novo nome (${socio.nome}): `, nome => {
        rl.question(`Novo email (${socio.email}): `, email => {
          rl.question(`Nova senha: `, senha =>{
            socio.nome = nome || socio.nome;
            socio.email = email || socio.email;
            socio.senha = senha || socio.senha;
            salvarSocio(socios);
            console.log('Socio editado.');
            cb();
          });
        });
      });
  });
}

function removerSocio(socios, rl, cb) {
  rl.question('ID do socio para remover: ', idStr => {
    const id = parseInt(idStr);
    const novosSocios = socios.
    filter(c => c.id !== id);
    if (novosSocios.length === socios.lenght) {
      console.log('Socio não encontrado.');
    } else {
      salvarSocio(novosSocios);
      console.log('Socio removido.');
    }
    cb();
  });
}

function menu() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  function loop() {
    const socios = carregarSocio();
    console.log('\n1. Listar\n2. Adicionar\n3. Editar\n4. Remover\n5. Sair');
    rl.question('Escolha: ', op => {
      if (op === '1'){
      listarSocio(socios, loop);
      } else if (op === '2') {
        addSocio(socios, rl, loop);
      } else if (op === '3') {
        editarSocio(socios, rl, loop);
      } else if (op === '4') {
        removerSocio(socios, rl, loop);
      } else if (op === '5') {
        rl.close();
      } else {
        console.log('Opção inválida');
        loop();
      }
    });
  }
  loop();
}

menu();

File: src/api/crudSocio.py
import json
import os

ARQUIVO = 'socios.json'

def carregar_socios():
    if not os.path.exists(ARQUIVO):
        return []
    with open(ARQUIVO, 'r', encoding='utf-8') as f:
        return json.load(f)

def salvar_socios(socios):
    with open(ARQUIVO, 'w', encoding='utf-8') as f:
        json.dump(socios, f, indent=2, ensure_ascii=False)

def listar_socios(socios):
    print('Lista de sócios:')
    for c in socios:
        print(f"ID: {c['id']} | nome: {c['nome']} | email: {c['email']}")

def adicionar_socio(socios):
    nome = input('Nome: ')
    email = input('Email: ')
    senha = input('Senha: ')
    novo_id = max([c['id'] for c in socios], default=0) + 1
    socios.append({'id': novo_id, 'nome': nome, 'email': email, 'senha': senha})
    salvar_socios(socios)
    print('Sócio adicionado.')

def editar_socio(socios):
    try:
        id_editar = int(input('ID do sócio para editar: '))
    except ValueError:
        print('ID inválido.')
        return
    socio = next((c for c in socios if c['id'] == id_editar), None)
    if not socio:
        print('Sócio inexistente.')
        return
    nome = input(f"Novo nome ({socio['nome']}): ") or socio['nome']
    email = input(f"Novo email ({socio['email']}): ") or socio['email']
    senha = input("Nova senha: ") or socio['senha']
    socio.update({'nome': nome, 'email': email, 'senha': senha})
    salvar_socios(socios)
    print('Sócio editado.')

def remover_socio(socios):
    try:
        id_remover = int(input('ID do sócio para remover: '))
    except ValueError:
        print('ID inválido.')
        return socios
    novos_socios = [c for c in socios if c['id'] != id_remover]
    if len(novos_socios) == len(socios):
        print('Sócio não encontrado.')
    else:
        salvar_socios(novos_socios)
        print('Sócio removido.')
    return novos_socios

def menu():
    while True:
        socios = carregar_socios()
        print('\n1. Listar\n2. Adicionar\n3. Editar\n4. Remover\n5. Sair')
        op = input('Escolha: ')
        if op == '1':
            listar_socios(socios)
        elif op == '2':
            adicionar_socio(socios)
        elif op == '3':
            editar_socio(socios)
        elif op == '4':
            socios = remover_socio(socios)
        elif op == '5':
            break
        else:
            print('Opção inválida')

if __name__ == '__main__':
    menu()
